resolve.go

```diff
@@ -107,3 +108,67 @@ func LookupServer(serverName ServerName) (*DNSResult, error) { // nolint: gocycl
 
 	return &result, nil
 }
+
+// ResolveServer implements the server name resolution algorithm described at
+// https://matrix.org/docs/spec/server_server/r0.1.1.html#resolving-server-names
+// Returns a slice containing the hosts (using the host:port form) that can be
+// used to send a federation request to the server using a given server name.
+// needWellKnown indicates whether a .well-known file should be looked up.
```

Comment:
why would this be false?

Reply:
Same reason Synapse does it that way (iirc): when it gets a result for a .well-known lookup, it calls itself with it because the rest of the steps are the same as when we want to resolve a server name, except we must skip the .well-known lookup.

Reply:
oh! hah. yes, sorry, I hadn't seen that.

Still, it would be nice to hide it from the public interface. Maybe make ResolveServer a thin wrapper around `resolveServer` which just sets needWellKnown.

Reply:
Sure, good idea.